src/components/BlogPostViewer.tsx

```diff
@@ -0,0 +1,91 @@
+import type { BlogPost, User, Comment } from "@prisma/client";
+import { useState, useMemo, useEffect } from "react";
+import { useInView } from "react-intersection-observer";
+import { api } from "~/utils/api";
+import { CommentModal } from "./CommentModal";
+import { Spinner } from "./Spinner";
+import { BlogPost as BlogPostComponent } from "./BlogPost";
+
+type Post = BlogPost & { user: User; comments: (Comment & { user: User })[] };
+
+export const BlogPostViewer: React.FC = () => {
+  const [view, setView] = useState<string>("Recent");
+  const { data: blogPosts, fetchNextPage } = api.blogPost.get.useInfiniteQuery(
+    {
+      take: 10,
```

Comment:
Just curious, does 10 posts fill the page?

Reply:
Yes it does from testing unless the screen size is really big, perhaps we increase it to 15/20 just to be safe?

Reply:
Its ok, I was just curious